### PR TITLE
feat: add salesforce triggers

### PR DESCRIPTION
**Pull request by abuaboud**
## What does this PR do?

New Record
Updated Record

Fixes # (issue)

## Type of change
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: packages/pieces/salesforce/src/lib/common/index.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                return {
+                    disabled: true,
+                    placeholder: 'connect your account first',
+                    options: [],
+                }
+            }   
+            const options = await getSalesforceObjects(value['authentication'] as OAuth2PropertyValue);
+            return {
+                disabled: false,
+                options: options.body['sobjects'].map((object: any) => {
```
Comment (khaledmashaly) [inline packages/pieces/salesforce/src/lib/common/index.ts:29]: Just to confirm, there's no typo in `sobjects`, right?

File: packages/pieces/salesforce/src/lib/trigger/new-updated-record.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    SELECT
+      FIELDS(ALL)
+    FROM
+      ${object}
+    WHERE LastModifiedDate > ${startDate}
+    ORDER BY LastModifiedDate ASC
+    LIMIT ${limit}
+    OFFSET ${offset}
+  `;
+}
```
Comment (khaledmashaly) [inline packages/pieces/salesforce/src/lib/trigger/new-updated-record.ts:92]: please note that `constructQuery` and `getRecords` methods are implemented in both triggers

File: packages/pieces/salesforce/project.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"name":"pieces-salesforce","$schema":"../../../node_modules/nx/schemas/project-schema.json","sourceRoot":"packages/pieces/salesforce/src","projectType":"library","targets":{"build":{"executor":"@nrwl/js:tsc","outputs":["{options.outputPath}"],"options":{"outputPath":"dist/packages/pieces/salesforce","tsConfig":"packages/pieces/salesforce/tsconfig.lib.json","packageJson":"packages/pieces/salesforce/package.json","main":"packages/pieces/salesforce/src/index.ts","assets":["packages/pieces/salesforce/*.md"],"buildableProjectDepsInPackageJsonType":"dependencies"}},"lint":{"executor":"@nrwl/linter:eslint","outputs":["{options.outputFile}"],"options":{"lintFilePatterns":["packages/pieces/salesforce/**/*.ts"]}}},"tags":[]}
```
Comment (khaledmashaly) [inline packages/pieces/salesforce/project.json]: I think it's better if this file is formatted

**Review by khaledmashaly — DISMISSED** [review]

File: packages/pieces/salesforce/src/lib/common/index.ts (hunk 1 above)
Comment (abuaboud) [reply packages/pieces/salesforce/src/lib/common/index.ts:29]: salesforces objeects ;)

**Pull request merged**